src/main.cpp

```diff
@@ -31,183 +31,224 @@ using TIME = NDTime;
+                                                                         ics_Network 
+                                                                         );
+
+    /************************/
+    /*******ABPSimulator********/
+    /************************/
+    cadmium::dynamic::modeling::Ports iports_ABPSimulator = {typeid(inp_control)};
+    cadmium::dynamic::modeling::Ports oports_ABPSimulator = {typeid(outp_ack),typeid(outp_pack)};
+    cadmium::dynamic::modeling::Models submodels_ABPSimulator = {sender1, receiver1,NETWORK};
+    cadmium::dynamic::modeling::EICs eics_ABPSimulator = {
+        cadmium::dynamic::translate::make_EIC<inp_control, sender_defs::controlIn>("sender1")
+    };
+    cadmium::dynamic::modeling::EOCs eocs_ABPSimulator = {
+        cadmium::dynamic::translate::make_EOC<sender_defs::packetSentOut,outp_pack>("sender1"),
+        cadmium::dynamic::translate::make_EOC<sender_defs::ackReceivedOut,outp_ack>("sender1")
+    };
+    cadmium::dynamic::modeling::ICs ics_ABPSimulator = {
+        cadmium::dynamic::translate::make_IC<sender_defs::dataOut, inp_1>("sender1","Network"),
+        cadmium::dynamic::translate::make_IC<outp_2, sender_defs::ackIn>("Network","sender1"),
+        cadmium::dynamic::translate::make_IC<receiver_defs::out, inp_2>("receiver1","Network"),
+        cadmium::dynamic::translate::make_IC<outp_1, receiver_defs::in>("Network","receiver1")
+    };
+    std::shared_ptr<cadmium::dynamic::modeling::coupled<TIME>> ABPSimulator =
+                                                                              std::make_shared<cadmium::dynamic::modeling::coupled<TIME>>(
+                                                                              "ABPSimulator", 
+                                                                              submodels_ABPSimulator, 
+                                                                              iports_ABPSimulator, 
+                                                                              oports_ABPSimulator, 
+                                                                              eics_ABPSimulator, 
+                                                                              eocs_ABPSimulator, 
+                                                                              ics_ABPSimulator 
+                                                                              );
+
+    /************************/
+    /*******TOP MODEL********/
+    /************************/
+    cadmium::dynamic::modeling::Ports iports_TOP = {};
+    cadmium::dynamic::modeling::Ports oports_TOP = {typeid(outp_pack),typeid(outp_ack)};
+    cadmium::dynamic::modeling::Models submodels_TOP = {generator_con, ABPSimulator};
+    cadmium::dynamic::modeling::EICs eics_TOP = {};
+    cadmium::dynamic::modeling::EOCs eocs_TOP = {
+       cadmium::dynamic::translate::make_EOC<outp_pack,outp_pack>("ABPSimulator"),
```

Comment:
Check for indentation in this line and next line.

Reply:
Done as requested, Thanks